ballerina/schema-client.bal

```diff
@@ -0,0 +1,320 @@
+    #
+    # + host - The name of where the postgresql server is hosted (ex: localhost)
+    # + user - The username to access the database
+    # + password - The password to access the database
+    # + database - The name of the database to be accessed
+    # + port - The port the database will be accessed on
+    # + options - PostgresSql database options
+    # + connectionPool - The `sql:ConnectionPool` to be used for the connection. If there is no
+    #                    `connectionPool` provided, the global connection pool (shared by all clients) will be used                  
+    # + return - An `sql:Error` or `()`
+    public function init(string host, string user, string password, string database, int port, 
+            Options? options = (), sql:ConnectionPool? connectionPool = ()) returns sql:Error? {        
+        self.database =  database;
+        self.dbClient = check new (host, user, password, database, port, options, connectionPool);
+    }
+
+    # Retrieves all tables in the database.
+    #
+    # + return - A string array containing the names of the tables or an `sql:Error`
+    isolated remote function listTables() returns string[]|sql:Error {
+        string[] tables = [];
+        stream<record {}, sql:Error?> results = self.dbClient->query(
+            `SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES 
+            WHERE TABLE_CATALOG = ${self.database} AND TABLE_SCHEMA NOT IN ('pg_catalog', 'information_schema');`
+        );
+
+        do {
+            tables = check from record {} result in results
+                select <string>result["table_name"];
+        } on fail error e {
+            return error sql:Error(string `Error while listing the tables in the ${self.database} database.`, cause = e);
+        }
+
+        check results.close();
+
+        return tables;
+    }
+
+    # Retrieves information relevant to the provided table in the database.
+    #
+    # + tableName - The name of the table
+    # + include - Options on whether column and constraint related information should be fetched.
+    #             If `NO_COLUMNS` is provided, then no information related to columns will be retrieved.
+    #             If `COLUMNS_ONLY` is provided, then columnar information will be retrieved, but not constraint
+    #             related information.
+    #             If `COLUMNS_WITH_CONSTRAINTS` is provided, then columar information along with constraint related
+    #             information will be retrieved
+    # + return - An 'TableDefinition' with the relevant table information or an `sql:Error`
+    isolated remote function getTableInfo(string tableName, sql:ColumnRetrievalOptions include = sql:COLUMNS_ONLY) returns TableDefinition|sql:Error {
+        record {}|sql:Error 'table = self.dbClient->queryRow(
+            `SELECT TABLE_TYPE FROM INFORMATION_SCHEMA.TABLES 
+             WHERE (TABLE_CATALOG=${self.database} AND TABLE_NAME = ${tableName});`
```

Comment:
What would happen if there are two tables with the same name in two different schema?

Reply:
Given that we do not have the schema, I assume that it will only utilize one of the tables instead of both